src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs

```diff
@@ -160,9 +160,9 @@ private void RequestViewRefreshHandler()
             View.InvalidateRender();
         }
 
-        private void RequestCreateModelsHandler(IEnumerable<IRenderPackage> packages)
+        private void RequestCreateModelsHandler(IEnumerable<IRenderPackage> packages, bool forceAsyncCall = false)
         {
-            if (CheckAccess())
+            if (!forceAsyncCall && CheckAccess())
             {
                 ViewModel.GenerateViewGeometryFromRenderPackagesAndRequestUpdate(packages);
             }
```

Comment:
Thanks @aparajit-pratap, as discussed I'm not quite comfortable with the way we add another parameter here to allow the method to be used in different ways. The existing implementation internally chooses to execute the `RequestUpdate` synchronous or asynchronously depending on the calling thread, that itself is already a hidden thing from the caller. I'd much rather not make this behavior public by adding that parameter.

As discussed also is that we can try to make this method to always call `RequestUpdate` asynchronously, thus removing the need of `forceAsyncCall` parameter. Since that involves a much larger design decision, we'll place this on old. Thanks again!